### PR TITLE
Fix ukraine source and add flake8 to reqs

### PR DESCRIPTION
**Pull request by robredpath**
The Ukraine source was erroneously downloading all of the files, when it's only the most recent we want.

### CODE REVIEW

**Comment by odscjames** [comment]
I'm concerned sample mode might be broken in that it might download old data?

I suspect the "if sample blah blah" line in gather_all_download_urls means that in sample mode, it would only get the first 3 and then you would end up taking the last one of those and end up downloading old data?

I suspect the fix is take the "if sample blah blah" line in gather_all_download_urls out and leave the "if sample blah blah" line in save_url in?

**Comment by odscjames** [comment]
Rest looks fine tho!

**Comment by odscjames** [comment]
Note this P.R. now covers both Rob's original comment in this P.R. and https://github.com/open-contracting/ocdsdata/issues/102

**Comment by jpmckinney** [comment]
Re: #102, I wonder whether arguments from sources should be prefixed in the cli with the `source_id`, to ensure uniqueness. I also wonder whether `argument_definitions` can be a dict, with each key being an argument name, and each value being a dict, which would serve as the keyword arguments to the `add_argument` call. (We can also pass the keyword arguments as e.g. `**kwargs` instead of explicitly, to allow sources to take advantage of features like `type` etc.)

**Comment by odscjames** [comment]
One thing I was aware of when designing this is that while it's currently true the CLI is the only way these are called, that might not be true in the future (we are already advertising this as a library). So for example I didn't just pass the parser object to the source (an early design), as that would tie it to the CLI to much. I think the dict & **kwargs suggestion would do the same. 

I'm not happy that I've totally got to an ideal state of separation, but I felt it was good enough as a first pass. I'm also wary of generalizing to much before we have several use cases (at the moment we only have one tiny one) - indeed I can already think of some complications if this feature was more heavily used later, but we really have to wait and see what real use is before deciding those. (Similar with unique issues).

File: requirements.txt [hunk 1]
```diff
@@ -19,4 +19,5 @@ alembic==0.9.9
 ratelimit==1.4.1
 lxml==4.2.1
 feedparser==5.2.1
+flake8==3.5.0
```
Comment (robredpath) [inline requirements.txt]: Do we need to introduce a requirements_dev ?

Comment (odscjames) [reply requirements.txt]: Maybe, but you added that to requirements originally :smile: Do you want to add a new commit to this P.R. to fix or leave this for now?

Comment (tian2992) [reply requirements.txt]: The flake part discussion should probably be involved in #112

**Pull request merged**